lib/parse.js

```diff
@@ -184,14 +193,19 @@ Parse.prototype._readFile = function () {
 
         return new Promise(function(resolve, reject) {
           self.stream(eof)
-            .pipe(inflater)
-            .on('error',function(err) { self.emit('error',err);})
-            .pipe(entry)
-            .on('finish', function() {
-              return fileSizeKnown ?
-                self._readRecord().then(resolve).catch(reject) :
-                self._processDataDescriptor(entry).then(resolve).catch(reject);
-            });
+          .pipe(inflater)
+          .on('pause', function() {
+            return fileSizeKnown ?
+            self._readRecord().then(resolve).catch(reject) :
+            self._processDataDescriptor(entry).then(resolve).catch(reject);
+          })
+          .on('error',function(err) {self.emit('error',err);})
+          .pipe(entry)
+          .on('finish', function() {
+            return fileSizeKnown ?
+              self._readRecord().then(resolve).catch(reject) :
+              self._processDataDescriptor(entry).then(resolve).catch(reject);
```

Comment:
```suggestion
            return fileSizeKnown
              ? self._readRecord().then(resolve).catch(reject)
              : self._processDataDescriptor(entry).then(resolve).catch(reject);
```

Reply:
@matthewsecrist  I can make these format changes if you feel strongly, I was just trying to match the code in the original lib (which doesn't follow our eslint rules at all obviously)?

Reply:
I don't mind if we don't make these adjustments. Just figured I'd make the suggestion. :)